test/Feature/ImplicitBindings/simple-gaps.test

```diff
@@ -0,0 +1,75 @@
+#--- source.hlsl
+
+RWBuffer<int> A : register(u1);
+RWBuffer<int> B;                 // gets u0
+RWBuffer<int> C: register(u2);   // unused
+RWBuffer<int> D;                 // gets u2
+
+cbuffer CB { // gets b0
+  int a;
+}
+
+// TEST GAPS:
+// Need to add tests for:
+// - arrays of resources (array descriptors are not yet supported by the offload test suite)
+// - resources in user-defined structs
+// - combination of the above
```

Comment:
I think there was some confusion over my use of the word "gaps" as a suggestion for the test file name. I was suggesting that this test is testing gaps in implicit bindings (ie, we have explicit bindings for A and C, but an implicit binding in between them with B).

The tests that fill the gaps in testing that you mention here would go in their own files (arrays.test, in-structs.test, etc), and so a comment in this file about them doesn't really make a lot of sense. Better to file issues saying we should write those tests instead.

Reply:
Got it :) In a way implicit bindings are always trying to fill gaps, even if the gap is 0 to UINT_MAX.

The tasks to add tests for arrays and structs are llvm/wg-hlsl#175 and llvm/wg-hlsl#205.